Denis_Protsenko.py

```diff
@@ -0,0 +1,218 @@
+from pydantic import BaseModel
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from typing import Union
+
+
+# Абстрактний клас
+class PublicModel(ABC):
+    @abstractmethod
+    def book_info(self):
+        pass
+
+
+# Клас моделі книги
+class BookModel(BaseModel):
+    title: str
+    author: str
+    year: int
+
+
+# Клас моделі журналу
+class MagazineModel(BaseModel):
```

Comment:
You can inherit from from BookModel, because all fields are the same
Or you can create abstract BaseModel which will have this poroperties, and inherit BookModel and MagazineModel from it.
```
import abc

from pydantic import BaseModel


class FooBarModel(BaseModel, abc.ABC):
    a: str
    b: int

    @abc.abstractmethod
    def my_abstract_method(self):
        pass

```

https://docs.pydantic.dev/latest/concepts/models/